source/main/gfx/EnvironmentMap.cpp

```diff
@@ -201,13 +201,9 @@ RoR::GfxEnvmap::~GfxEnvmap()
         if (m_cameras[face] != nullptr)
         {
             gEnv->sceneManager->destroyCamera(m_cameras[face]);
+            m_render_targets[face]->removeAllViewports();
         }
     }
-
-    if (!m_rtt_texture.isNull())
-    {
-        Ogre::TextureManager::getSingleton().remove(m_rtt_texture->getName());

```

Comment:
I'm fairly sure removing this will cause issues when you return to menu and spawn a different map (previous terrain reflected, no updates), that's why I added the cleanup IIRC.

Reply:
Can't we reuse the old texture? The problem is that we cannot create textures in the EnvMap constructor anymore, or can we?

Reply:
Ah, yes, we can: https://github.com/RigsOfRods/rigs-of-rods/pull/1537/files#diff-7baed2733bb2b4085178f51abfb8b32bR45

Reply:
So, do we have an issue or not?

Reply:
Me neither, because you cannot spawn a different map right now without crashing.